src/Sylius/Bundle/ProductBundle/Doctrine/ORM/ProductVariantRepository.php

```diff
@@ -177,4 +177,18 @@ public function getCodesOfAllVariants(): array
             ->getArrayResult()
         ;
     }
+
+    public function findDefaultVariant(ProductInterface $product, bool $enabled = true): ?ProductVariantInterface
+    {
+        return $this->createQueryBuilder('o')
+            ->innerJoin('o.translations', 'translation')
+            ->andWhere('o.product = :product')
+            ->andWhere('o.enabled = :enabled')
+            ->setParameter('product', $product)
+            ->setParameter('enabled', $enabled)
+            ->addOrderBy('o.position', 'ASC')
+            ->getQuery()
```

Comment:
Won't `getQuery()->getResult()` return array?

Reply:
oh yes you are right, I was about to test the PR and then read the PR suggestion. But anyway I reverted this reprository method to the `findOneBy` instead to avoid BC break.